include/seqan3/std/concepts

```diff
@@ -86,7 +84,7 @@ using ::concepts::unsigned_integral;
 // using ::concepts::floating_point;
 //!\cond
 template <typename t>
-SEQAN3_CONCEPT floating_point = std::is_floating_point_v<t>;
+META_CONCEPT floating_point = std::is_floating_point_v<t>;
```

Comment:
where does the META_CONCEPT come from ?

Reply:
From the ranges library

Reply:
> So we don't need any of the stuff defined in platform in the std headers?

In C++10 mode, we also don't import any seqan3 code, so we should make those headers agnostic of them.